### PR TITLE
fix iBundleProvider.getResources(from,to) throwing a NPE

### PR DESCRIPTION
**Pull request by fil512**

### CODE REVIEW

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/api/server/IBundleProvider.java [hunk 1]
```diff
@@ -113,6 +115,26 @@ default String getPreviousPageId() {
...
 	List<IBaseResource> getResources(int theFromIndex, int theToIndex);
 
+	/**
+	 * Get all resources
+	 *
+	 * @return getResources(0, this.size ()).  Return an empty list if size() is zero.
+	 * @throws ConfigurationException if size() is null
+	 */
+	@Nonnull
+	default List<IBaseResource> getResources() {
```
Comment (jamesagnew) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/api/server/IBundleProvider.java:125]: Maybe `getAllResources` would be a more descriptive name?

**Review by jamesagnew — APPROVED** [review]
Looks good

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/api/server/IBundleProvider.java (hunk 1 above)
Comment (fil512) [reply hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/api/server/IBundleProvider.java]: Good call.  Done.

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/api/server/IBundleProvider.java [hunk 2]
```diff
@@ -113,6 +115,26 @@ default String getPreviousPageId() {
...
 	List<IBaseResource> getResources(int theFromIndex, int theToIndex);
 
+	/**
+	 * Get all resources
+	 *
+	 * @return getResources(0, this.size ()).  Return an empty list if size() is zero.
+	 * @throws ConfigurationException if size() is null
+	 */
+	@Nonnull
+	default List<IBaseResource> getAllResources() {
```
Comment (tadgh) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/api/server/IBundleProvider.java:125]: what a nice change this will be. 👍🏻

**Pull request merged**